packages/data-store/src/contact/ContactStore.ts

```diff
@@ -276,96 +262,247 @@ export class ContactStore extends AbstractContactStore {
+    })
+
+    const createdResult: ContactRelationshipEntity = await (await this.dbConnection).getRepository(ContactRelationshipEntity).save(relationship)
+
+    return contactRelationshipFrom(createdResult)
   }
 
-  private identityFrom = (identity: IdentityEntity): IIdentity => {
-    return {
-      id: identity.id,
-      alias: identity.alias,
-      roles: identity.roles,
-      identifier: this.correlationIdentifierFrom(identity.identifier),
-      ...(identity.connection && { connection: this.connectionFrom(identity.connection) }),
-      metadata: identity.metadata ? identity.metadata.map((item: IdentityMetadataItemEntity) => this.metadataItemFrom(item)) : [],
-      createdAt: identity.createdAt,
-      lastUpdatedAt: identity.createdAt,
+  getRelationship = async ({ relationshipId }: IGetRelationshipArgs): Promise<IContactRelationship> => {
+    const result: ContactRelationshipEntity | null = await (await this.dbConnection).getRepository(ContactRelationshipEntity).findOne({
+      where: { id: relationshipId },
+    })
+
+    if (!result) {
+      return Promise.reject(Error(`No relationship found for id: ${relationshipId}`))
     }
+
+    return contactRelationshipFrom(result)
   }
 
-  private correlationIdentifierFrom = (identifier: CorrelationIdentifierEntity): ICorrelationIdentifier => {
-    return {
-      id: identifier.id,
-      type: identifier.type,
-      correlationId: identifier.correlationId,
+  getRelationships = async (args?: IGetRelationshipsArgs): Promise<Array<IContactRelationship>> => {
+    const contactRelationshipRepository: Repository<ContactRelationshipEntity> = (await this.dbConnection).getRepository(ContactRelationshipEntity)
+    const initialResult: Array<ContactRelationshipEntity> | null = await contactRelationshipRepository.find({
+      ...(args?.filter && { where: args?.filter }),
+    })
+
+    const result: Array<ContactRelationshipEntity> | null = await contactRelationshipRepository.find({
+      where: {
+        id: In(initialResult.map((contactRelationship: ContactRelationshipEntity) => contactRelationship.id)),
+      },
+    })
+
+    return result.map((contactRelationship: ContactRelationshipEntity) => contactRelationshipFrom(contactRelationship))
+  }
+
+  updateRelationship = async ({ relationship }: IUpdateRelationshipArgs): Promise<IContactRelationship> => {
+    const contactRelationshipRepository: Repository<ContactRelationshipEntity> = (await this.dbConnection).getRepository(ContactRelationshipEntity)
+    const result: ContactRelationshipEntity | null = await contactRelationshipRepository.findOne({
+      where: { id: relationship.id },
+    })
+
+    if (!result) {
+      return Promise.reject(Error(`No contact relationship found for id: ${relationship.id}`))
+    }
+
+    const contactRepository: Repository<ContactEntity> = (await this.dbConnection).getRepository(ContactEntity)
+    const leftContact: ContactEntity | null = await contactRepository.findOne({
```

Comment:
Fetching both the left and right sides of a relationship just to check whether they exist is done in multiple locations, so probably better to just create a method out of it

Reply:
refactored to a function